collector/corosync/parser.go

```diff
@@ -0,0 +1,213 @@
+package corosync
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type Parser interface {
+	Parse(cfgToolOutput []byte, quorumToolOutput []byte) (*Status, error)
+}
+
+type Status struct {
+	NodeId      string
+	RingId      string
+	Seq         uint64
+	Rings       []Ring
+	QuorumVotes QuorumVotes
+	Quorate     bool
+	Members     []Member
```

Comment:
my personal pov is having nested struct looks more compact rather then all this sparsed structs . but is more taste. I prefer however compacted struct they are easier to read, instead of jumping around to mini-struct tied together.

Reply:
yeah, that's definitely very subjective: I personally don't like nested structures, as you can't type hint to them individually.

Reply:
ok it was just I wanted to highlight in sake, I will do a pr when you sleep
so I can change it again.. Joking :grin: